src/fmeval/constants.py

```diff
@@ -20,6 +20,9 @@
 BUILT_IN_DATASET_PREFIX = "s3://fmeval/datasets"
 BUILT_IN_DATASET_DEFAULT_REGION = "us-west-2"
 
+ISOF_BUILT_IN_DATASET_REGION = "us-isof-south-1"
```

Comment:
Let's create a mapping from iso region -> iso region that holds the builtin datasets so that we can easily extend the code to support other iso regions in the future.
```
BUILT_IN_DATASET_ISO_REGIONS = {
    "us-isof-south-1": "us-isof-south-1",
    "us-isof-east-1": "us-isof-south-1"
}
ISO_REGIONS = ["us-isof-south-1", "us-isof-east-1"]  # to be extended with other iso regions later
```